src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlException.cs

```diff
@@ -146,7 +146,7 @@ public override string ToString()
             sb.AppendFormat(SQLMessage.ExClientConnectionId(), _clientConnectionId);
 
             // Append the error number, state and class if the server provided it
-            if (Number != 0)
+            if (Errors.Count > 0 && Number != 0)
```

Comment:
Do we need `Errors.Count > 0` since it's already checked in the `Number` property?

Reply:
To not append line in that case... the check is getters are to ensure getters can function, while this is mainly for formatting message appropriately.